### PR TITLE
Benchmark with DaCe cpu and gpu backends

### PR DESCRIPTION
**Pull request by fmalatino**
## Purpose

Branch for benchmarking of PACE. It includes four new configuration files to be used in benchmarking with DaCe backends for use with cpus and gpus.

## Code changes:

- Addition of four configuration files: baroclinic_c384_cpu/gpu.yaml, baroclinic_c3072_cpu/gpu.yaml
- Use of dace_inhibitor decorator in microphysics.py

## Requirements changes:

- In constraints.txt: dace updated to 0.15.1
- In constraints.txt: gridtools-cpp updated to 2.3.1

## Infrastructure changes:

- N/A

## Checklist
Before submitting this PR, please make sure:

- [x] You have followed the coding standards guidelines established at [Code Review Checklist](https://drive.google.com/file/d/1R0nqOxfYnzaSdoYdt8yjx5J482ETI2Ft/view?usp=sharing).
- [x] Docstrings and type hints are added to new and updated routines, as appropriate
- [x] All relevant documentation has been updated or added (e.g. README, CONTRIBUTING docs)
- [x] For each public change and fix in `pace-util`, HISTORY has been updated
- [x] Unit tests are added or updated for non-stencil code changes

### CODE REVIEW

**Comment by FlorianDeconinck** [comment]
The DaCe version has been updated? Did we run model validation on it?

**Comment by fmalatino** [comment]
> The DaCe version has been updated? Did we run model validation on it?

I have to change it to your gcc_dies_on_dacecpu branch, will be in the coming push.

**Review by bensonr — APPROVED** [review]

File: .gitmodules [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [submodule "external/gt4py"]
 	path = external/gt4py
 	url = https://github.com/gridtools/gt4py.git
-[submodule "external/dace"]
+[submodule "dacecpufix"]
 	path = external/dace
-	url = https://github.com/spcl/dace.git
+	url = https://github.com/FlorianDeconinck/dace.git
```
Comment (oelbert) [inline .gitmodules:6]: I really do not love doing this, cannot wait to have it resolved

File: driver/examples/configs/baroclinic_c384_cpu.yaml [hunk 2]
```diff
@@ -72,7 +71,7 @@ dycore_config:
   nwat: 6
   p_fac: 0.1
   rf_cutoff: 800.
-  rf_fast: false
+  rf_fast: true
```
Comment (oelbert) [inline driver/examples/configs/baroclinic_c384_cpu.yaml:74]: Has rf_fast=True been implemented?

File: tests/main/driver/test_example_configs.py [hunk 3]
```diff
@@ -28,6 +28,10 @@
     "baroclinic_c12_orch_cpu.yaml",
     "tropical_read_restart_fortran.yml",
     "tropicalcyclone_c128.yaml",
+    "baroclinic_c384_cpu.yaml",
+    "baroclinic_c384_gpu.yaml",
+    "baroclinic_c3072_cpu.yaml",
+    "baroclinic_c3072_gpu.yaml",
```
Comment (oelbert) [inline tests/main/driver/test_example_configs.py:34]: Should these be included in the PR?

**Review by oelbert — COMMENTED** [review]
Mostly fine, just a couple of questions to clear up

File: tests/main/driver/test_example_configs.py (hunk 3 above)
Comment (oelbert) [reply tests/main/driver/test_example_configs.py:34]: actually, I think the yaml configs should be in the PR but I don't think we want to include such a high-res run in the test framework, and I'm not sure about the c384 ones either, actually

Comment (bensonr) [reply tests/main/driver/test_example_configs.py:34]: @oelbert - this PR is going to a _benchmark_ branch for use when conducting procurements.

File: driver/examples/configs/baroclinic_c384_cpu.yaml (hunk 2 above)
Comment (fmalatino) [reply driver/examples/configs/baroclinic_c384_cpu.yaml:74]: It has not, but when run with it set to false, it throws an NotImplementedError when tau != 0. Should this instead be removed or tau == 0?

Comment (bensonr) [reply driver/examples/configs/baroclinic_c384_cpu.yaml:74]: You could cherry pick over the not-implemented PR from Florian and remove the options from the yaml config.

File: tests/main/driver/test_example_configs.py (hunk 3 above)
Comment (oelbert) [reply tests/main/driver/test_example_configs.py:34]: Hah, I'd missed that, thanks for flagging it!

File: driver/examples/configs/baroclinic_c384_cpu.yaml (hunk 2 above)
Comment (oelbert) [reply driver/examples/configs/baroclinic_c384_cpu.yaml:74]: I think I'd rather change tau than set something in the config that we haven't actually implemented in the model, at least for now. I'll create an issue for it as well though so we can implement a config that's internally consistent.

**Review by oelbert — APPROVED** [review]
👍

**Pull request merged**